Gives humans two types of mutant accessories: ears and (human-specific) tails.

Ricotez opened the pull request:
<h1>Because of the extremely controversial nature of this PR, I've removed roundstart access to ears and tails. The rest of the framework is still in place, but this is an admin-only thing now.</h1>


Expanding upon the system recently added to lizards, humans can now set two of their own mutant accessories: (human-specific) tails and ears.

I've only added one type of ear and one type of tail to this first pull request, my hope is that more talented spriters will supply new types. The ears and tail are taken from the kitty ears item in-game; while the ears are clearly those of a cat, the tail is generic enough to be used for other animals, such as monkey people.

Here's an example of a monkeyperson, a human with brown hair, a brown beard and a brown tail (but no ears):
![ss13monkeyperson](https://cloud.githubusercontent.com/assets/3974988/8393473/d33c80a6-1d10-11e5-9a4c-cf747c654d96.png)

And if you do set the ears, you end up with a catperson:
![ss13catperson](https://cloud.githubusercontent.com/assets/3974988/8393471/cc889830-1d10-11e5-84a0-30853aa2f932.png)

And yes, you can also wag the tails just like lizards can, with *wag and *stopwag.

<b>But Ricotez, why are the tails and ears the same colour as the hair?</b>
The above examples were taken when the colours were still decided by mutantcolour, I just chose those colours extremely similar to the hair colours. In their current version this particular tail and these particular ears use your hair colour instead, and I modified the system such that you can choose per mutant part which colour it should use.

<b>But Ricotez, now my random characters will be furries!</b>
The code that generates a random set of features will not randomize ears and human tails. So if you create a random human, you'll never get ears or a tail.

<b>But Ricotez, what if humans go run around with lizard tails now?</b>
I deliberately kept lizard and human tails separated from each other. They're even saved independently in your features list. Humans cannot set lizard tails and vice versa.

<b>But Ricotez, why do I always see a lizard tail in the setup preview now?</b>
I have absolutely no idea. There is something VERY strange going on with the setup preview, because whenever you set your tail, it will also render a lizard tail behind your south-looking preview sprite. The fact that it renders this alongside the cat/monkey tail (you can see it pop up behind your right shoulder in the south-looking sprite) and that it does not show up in the west-looking sprite only adds to my confusion. The amount of time I spent on trying to fix this one particular bug is bigger than the rest of the project took me, so eventually I just gave up. I think we can all agree that the code that generates the preview image is awful anyway.

<b>But Ricotez, why did you think this was a good idea?</b>
Why should lizards have all the fun of our new mutant accessory system?

<b>But Ricotez, I don't want this on my server!</b>
Access to these features is locked under the same config option as roundstart lizards. If you have the config option disabled, human players cannot set tails or ears in their character setup.

Commit:
-Added two new types of mutant accessories for humans: tail_human and ears. These ears are considered mutant parts and don't override normal human ears.
-Added one ear and one tail accessory to these categories, obtained from the kitty ears. The ears are obviously cat ears, but the tail can pass for a monkey tail without ears.
-Humans can now also edit alien/mutant colours. They have access to the full range of colours, but if the player switches back to lizard and the colour is too dark, it'll be reset to the default.
-Renamed the original tail to tail_lizard. All references are properly updated and nothing has to be changed.
-Could not solve an annoying bug with the setup window, where a lizard tail is rendered whenever a human tail is set. This problem only exists in the setup window and does not affect the sprite in-game.

Xhuis commented:
![](http://www.its.caltech.edu/~atomic/snowcrystals/photos/w031230a113.jpg)

GunHog commented:
MEW MEW MEW MEOW! :+1: I would rather have mutant races though, because you can change the saymod to "meows" and "purrs", and slash instead of punch. Either way, I love this, and every anti-lizard biggot is going to catch fire when they see this!

Incoming5643 commented:
When I said people could expand it easily to other mutant races / humans I meant like scars or rad tats or something.

I don't know how to feel about this, it's several steps past silly to me.

Dorsisdwarf commented:
:-1: 
No.

Thunder12345 commented:
As a professional catgirl I give this a :+1:
Yeah, preview code needs desperately rewritten at some point, I had to look in it once and it looks pretty awful.

Incoming5643 commented:
This is just a hunch but renaming tail_lizard and tail_human to lizard_tail and human_tail may fix your preview image woes.

Why? No idea, but I experienced something similar when I was fighting my way through it. No assurances it will work.

That said this is still a definite ![341s6](https://cloud.githubusercontent.com/assets/4176358/8393576/71973e54-1ce3-11e5-84ac-21e825b7057d.png) for being really silly and also enabling the furries.

Iamgoofball commented:
:-1:

Okay, no, this is going too far now. Take your stupid fetishes elsewhere.

MrPerson commented:
When goofball says you've gone too far

You've gone WAY too far

:-1:

Durkel commented:
The prelude to stuttering catbeasts

SHUT IT DOWN!

:-1:

Niknakflak commented:
Haha this is hilarious.  I think the only reason people freak out is because it's cat ears.  If it were dolphin fins persay, no one would have any issues.

demonfiren commented:
Personally, I prefer catbeasts to these abominations.

Chiefly because everyone knows catbeasts are the lowest scum on the station, but those even the AI still has to treat as human.

Iamgoofball commented:
if this gets in I will murder people using these unironically roundstart every round

Razharas commented:
# **EVERY DOLLAR SPENT ON ANYTHING IS A DOLLAR NOT SPENT ON RESEARCHING THE GENETICALLY MODIFIED CATGIRLS**
![image](http://iichan.hk/b/src/1427233379348.jpg)
![image](http://iichan.hk/b/src/1431090842686.jpg)
![image](http://iichan.hk/b/src/1430976236198.jpg)

Iamgoofball commented:
# DE MAINTAINER RAZHARAS WHEN

Saltycut commented:
no :-1:

Ricotez commented:
Get yer neko bullshit out of my beautiful PR raz.

Also I forgot to lock humans with mutant bodyparts out of command roles, I'm testing that right now and I'll update when it works.

vista- commented:
:-1:

Lizards are a fun gimmick, this is just way too much.

nukeop commented:
no, go away

Iamgoofball commented:
@Ricotez 
don't

WJohn commented:
I intended this to be expanded eventually to tattoos and piercings.

But... not like this.

NOT LIKE THIS.

Iamgoofball commented:
save the day wj

close the pr

we all want it

Razharas commented:
Also since erp is currently banned on servers theres little reason to merge this

Ricotez commented:
>implying I only want to merge this so people can get their erp fantasies on

WJohn commented:
We're not implying anything here when you're so clearly stating that with this PR :^)

Razharas commented:
I want this specifically because of my taste preferences dont judge

Ricotez commented:
if you only associate catpeople with erp fantasies then who is the one with the dirty mind here :^)

SconesC commented:
:-1: Enables furries

quaggaman555555 commented:
:+1: i no longer have to choose between ears or a space suit anymore, nya~!

Commit:
-Humans with tails and/or ears are considered not entirely human, and locked out of important positions.

quaggaman555555 commented:
Also, goddamnit this stupid app wont let me have raz's pictures!

Iamgoofball commented:
@quaggaman555555 fuck you furry

SconesC commented:
@quaggaman555555 into the garbage

Ricotez commented:
Humans with mutant bodyparts are not considered real humans by Nanotrasen and therefore fail the human authorization check now.

SconesC commented:
They still have server rule protections, not worth it

Fox-McCloud commented:
:+1:

Iamgoofball commented:
@fox-mccloud no

MMMiracles commented:
gas the nekos racewar now

dannno commented:
jesus christ

Thunder12345 commented:
Danno get back in your grave please.

Iamgoofball commented:
no thunder he stepped out of lifeweb to say that this is shit

ColonicAcid commented:
>lifeweb is now the big bad boogeyman

nice nice

WJohn commented:
I wouldn't associate them only with ERP if they were actually more than just that :)

ColonicAcid commented:
I was about to insult you but then I remembered the github tone police would remove my comment so you're lucky.

pudl commented:
no :-1:

palpatine213 commented:
:1: Best PR.

tedward1337 commented:
Oh my sweet summer child
Not like this

Iamgoofball commented:
@palpatine213 fuck you

@ColonicAcid no you fat fuk i was saying he came back just to tell this pr it sucked balls

Razharas commented:
Personal attacks please

Ricotez commented:
Stop shitting up my PR with ad hom or I'll add cat pun speech replacements in

duncathan commented:
:+1: for 3-way racewars

Fox-McCloud commented:
@WJohn 

What if they could have Xeno tails? =p

Falamazeer commented:
This is bad, and you should feel bad.

GunHog commented:
Is it possible to have "meows" or "purrs" for the saymods as relating to cat-people, or "chimpers" for monkey people?

Xhuis commented:
just make it so whenever a catperson talks they instantly die
this pr is fucking awful

PKPenguin321 commented:
what the literal fuck?
isn't this what people were joking would happen after the lizards thing and were posting in the *shitty ideas thread* about?
why is this a real PR?
what the fuck?
:-1:

Kromgar commented:
Ricotez fucking kill yourself.

NO.  WE ARE NOT DOING THIS FUCK FURRIES. LIZARDS ARE FAR ENOUGH. WE AINT GOING INTO FURRY TOWN

Ricotez commented:
wow ok no, death threats are where I draw the fucking line

I've had my laugh, close this fucking PR already

Razharas commented:
This pr cant get any more awsome than this

ike709 commented:
:+1: For managing to anger everyone that isn't a furry.

:-1: For the actual feature. It's just pointless in my opinion and generates too much drama.

Iamgoofball commented:
@Ricotez you can close it yourself you know

Xhuis commented:
chug chug goes the drama generator

Ricotez commented:
<img src="https://media0.giphy.com/media/JwNPAckJDiPsI/200_s.gif" />

ColonicAcid commented:
okay now that you got death threats i know that adding this would cause an amount of assblastery unbeknownst to humanity.

please readd this please ricotez i must have tears.

pudl commented:
ROUND ENDING KILL

RemieRichards commented:
bring it back.
we need this.

for good or bad we need this.

optimumtact commented:
wow you go away for 1 day and you miss all the best memes

Kromgar commented:
We will never need this

freerealestate commented:
Please reopen this, it pisses off all the right people.

On 28 June 2015 at 16:24, Kromgar <notifications@github.com> wrote:

> We will never need this
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tgstation/-tg-station/pull/10209#issuecomment-116191066>
> .
>

Tokiko1 commented:
Atleast add it to genetics and make the server option default to off.

mph55 commented:
What about the human ears? Will catpeople have four ears?

Ricotez commented:
I'm not going to reopen a pull request when it appears 90% of the community is against it, not even if it's just to piss them off

If you want me to reopen it, get people who actually want it added to post in here or in the feedback thread

Miauw62 commented:
I approve of this PR, if only to highlight how ridiculous and snowflakey
lizards are right now
Op 28-jun.-2015 11:51 schreef "Ricotez" <notifications@github.com>:

> I'm not going to reopen a pull request when it appears 90% of the
> community is against it, not even if it's just to piss them off
>
> If you want me to reopen it, get people who actually want it added to post
> in here or in the feedback thread
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tgstation/-tg-station/pull/10209#issuecomment-116249630>
> .
>

Thunder12345 commented:
Free catgirls 2015

CorruptComputer commented:
Everyone who wants this reopened say "Catgirls 2015"

Catgirls 2015

GunHog commented:
<b>Catgirls 2015"</b>

duncathan commented:
Catgirls 2015

freerealestate commented:
catgirl 420

On 29 June 2015 at 05:53, duncathan <notifications@github.com> wrote:

> Catgirls 2015
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tgstation/-tg-station/pull/10209#issuecomment-116303763>
> .
>

Thunder12345 commented:
Catgirls 4030/2

Miauw62 commented:
catgirls 2015

Xhuis commented:
# you are all furries

Miauw62 commented:
no i just need an excuse to remove snowflake lizard sprites

Xhuis commented:
that was fast

Miauw62 commented:
i was checking my email vOv

duncathan commented:
Tbh I just enjoy the racism roleplay

Bluespace commented:
i want to fuck catgirls

CorruptComputer commented:
@Bluespace Woah, calm down there.

RemieRichards commented:
@Xhuis 

Catgirls != furry rubbish
Catgirls = Cute

Catrgirls 2015.

palpatine213 commented:
Catgirls 2015

MMMiracles commented:
![catgrill2015](https://cloud.githubusercontent.com/assets/9276171/8398256/028e7392-1db5-11e5-83ad-b3ee203062e0.png)

Ricotez commented:
![outlawlayline3](https://cloud.githubusercontent.com/assets/3974988/8398463/3753b2ba-1def-11e5-82f6-9774b30441d6.png)


a rough tally says that the ration is about 50/50 now but I have seriously no fucking idea who is serious and who is trolling anymore

the whole thing is getting out of hand so I'll reopen the pull request and let the Gods (read: Headcoders) decide

please have me(w)rcy on my soul

duncathan commented:
:+1:

Cheridan commented:
wake me up inside

RemieRichards commented:
YAY.

Razharas commented:
>Day ago
>Its ready to merge

Ah i wonder what shall i do, hmmmmm

xxalpha commented:
nooooooooo :-1:

worse than economy

Iamgoofball commented:
NO

GODDAMN IT NO

Xhuis commented:
# ___REEEEEEEEEEEEEEEEEEEEEEEEEEEEE___

Chocobro commented:
catgirls 2015
(if anything this system can still be used, not just for tails. scars and freckles come to mind)

optimumtact commented:
http://www.xeon.pw/~francis/wakemeupinside.webm

GunHog commented:
YES!!!!!!!!!!!!!!!!!!!!!!!! Also Ctarl-Ctarls are my favorite animal girl!

ExcessiveUseOfCobblestone commented:
:-1: stick to begging the HoP for theatre access for the Kawaii cat ears.

:+1: as a base for things like scars/freckles/etc.

Razharas commented:
I want whole body prison tatoos so i can roleplay as tough russian gangsta

Ricotez commented:
problem with stuff like scars and tattoos is that they are not part of your genotype, at least in the real world

in the code however all mutant bodyparts are stored in your DNA so changelings would copy them over right now

I'd have to choose between game balance (changelings copying tattoos and scars too) or realism

daman997 commented:
never EVER :-1:

RemieRichards commented:
@Ricotez 
well scars is scar tissue, that's DNA, tattoos are ink, if this creature can emit EMP from it's vocal cords, turn into a monkey, form blades from it's arms, I'm sure it can produce ink sort of like a squid.

SconesC commented:
Just choose balance, it's not like genetics is terribly realistic

:+1: To more scars and tattoo things
:-1: Shameless furry enabling content

Ricotez commented:
I thought we effectively shut down furries when we banned ERP

also what the fuck is with all these labels

RemieRichards commented:
@Ricotez professional maintaining is occurring.

Incoming5643 commented:
I'm sorry. it's dumb and furry bait. People don't gotta be sexual to be REALLY WEIRD ABOUT IT.

KorPhaeron commented:
Can't be worse than roundstart lizards

Razharas commented:
>well scars is scar tissue, that's DNA

Jesus christ

ColonicAcid commented:
top tier biology here lads

Kromgar commented:
This will bring back the furries don't do it

ColonicAcid commented:
i'll take furries over you krom any day of the week

B-)

Kromgar commented:
U havin a gabber m8?

palpatine213 commented:
:+1: Merge Posthaste, never look back.

Tokiko1 commented:
Lizards have plenty of customization. Humans need some more too.

ike709 commented:
catgirls 2k15

Incoming5643 commented:
scar and tats, not tails and cats

:police_car: :police_car: :crying_cat_face: :gun: :cop: :police_car: :police_car:

Anonus commented:
DON'T DO IT

RemieRichards commented:
@Ricotez Cheridan wants elf ears, then he'll merge! :P

Cheridan commented:
add elf ears

Chocobro commented:
[00:32] (@Cheridan) i'm not merging it until he adds elf ears
:)

Ricotez commented:
I'll add elf <i>tears</i>

(I will gladly add all the ears you want but I need sprites)

Chocobro commented:
okay github is a cunt
@Cheridan | i'm not merging it until he adds elf ears

ToadStar commented:
I'll do elf ear sprites hold on.

Kromgar commented:
Don't do it.

I don't want fucking special snowflake simulator 13. Soon we'll start talking about triggers and eventually using bombs will be a bannable offense because they trigger Veterans of the Iraq war

MMMiracles commented:
itt: people over-react to sprites on a 2d-atmosphere simulator for a niche early 2000 engine.

Incoming5643 commented:
humans are supposed to be white bread, if you start blurring the line between mutantrace and human suddenly there's this whole grey area that's gonna be a real headache for AIs/Admins/Players

Just wear the damn cat ears please :(

PKPenguin321 commented:
end my life

freerealestate commented:
why ever care

On 29 June 2015 at 13:49, PKPenguin321 <notifications@github.com> wrote:

> end my life
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tgstation/-tg-station/pull/10209#issuecomment-116372833>
> .
>

Thunder12345 commented:
The nyaa approaches

Steelpoint commented:
On the off chance this is not all shitposting I vehemently disagree with this PR's stated changes, scars and whatnot are interesting but keep the mutations out of pre-round set up.

Thunder12345 commented:
You cannot stop the nyaapocalypse

RogerWilcoBeep commented:
Jesus fucking Christ, why?

Ricotez commented:
I've slightly rewritten the human authority check to now call a species-specific proc qualifies_for_rank(var/rank, var/list/features). This proc returns 1 by default, and must be overridden in the species subtype if you want to limit the number of jobs a species has access to at round start.

I left lizards the way they were, which means they only fail the check if the rank is a command role. But if a human has their ears or tail set to anything other than None, they will fail the check if the rank is in command, security, science, medical or engineering, or if it's QM. This means that mutants can only pick lower supply and civilian roles. (And of course silicon roles.)

Lizards have been around for much longer and had much more opportunities to prove their worth to Nanotrasen, while the company started hiring dirty mutants only recently. Mutants should be considered non-human as far as Nanotrasen and the silicons are concerned (similar to hulks), and treated just as lizards (or even worse).

I'll commit and push it as soon as I'm done testing.

Steelpoint commented:
I'm sorry but this is still a bad change to the game, we really don't need obnoxious, stereotypical and boring things like cat ears and what not.

This is not a change I think will be a good one for /tg/station. Lizards are already annoying enough to deal with in game let alone what this PR will unleash.

RemieRichards commented:
@Ricotez eh, I'd give them sec back out of all those removals.
if people who hate lizards let them into security, same should go for a halfhuman who's technically still half better than a lizard.

Ricotez commented:
@RemieRichards I think it's funnier if mutants are considered even lower than nonhumans like lizards, because that's pretty much what half the reactions to this PR have been like

Kromgar commented:
Mutants should be valid

Miauw62 commented:
snowflake lizards should be valid

On Mon, Jun 29, 2015 at 1:14 PM, Kromgar <notifications@github.com> wrote:

> Mutants should be valid
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tgstation/-tg-station/pull/10209#issuecomment-116613422>
> .
>

Miauw62 commented:
more seriously im sort of against this because the ridiculous amount of customization is making it literally impossible to disguise as anybody.

Ricotez commented:
@Miauw62 that's actually the best counterargument I've heard so far, short from stealing kitty ears it's almost impossible to disguise yourself as a mutant (although ears shouldn't render if your head item covers up your hair)

I can update ears and tails to use hair colour instead of mutant colour but that's only a partial fix

Razharas commented:
Inability to disguse as lizard is much more apparent really

Supermichael777 commented:
honestly the inability to disguise as this is offset by the job restrictions. however a fake mutant race accessory set with a inbuilt color wheel in the uplink would be cool maybe disguised car girl ears?

Kromgar commented:
Oh god no not traitor colored cat ears

Dorsisdwarf commented:
:-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1:

Ricotez commented:
I opened the pull request and it got overwhelmed by "NO GO FUCK YOURSELF :-1:  :-1:  :-1: "

I closed the pull request and it got overwhelmed by "#CATGIRL2015 :+1:  :+1:  :+1: "

a few people made good points but the number of comments with actual <i>feedback</i> in them rather than a fallacy or meme is only a fraction of the total. I don't ignore those comments but I only use them for general estimates on how the opinions in the community are divided, because that's all I can really do with them.

I closed the pull request because at first it appeared that the vast majority of the community was against it, then I reopened it when it suddenly got a lot of support. at this point I'm sick of being in the middle of everything, that's why I'm just pushing changes that I think should still be made and have coderbus decide.

Kromgar commented:
Well I think Miauw note about how its harder to disguise people is a damn good reason not to have this shit. Also we are adding too much special snowflake shit and getting into furry territory. That I do not want

GunHog commented:
Erro's Law is in full force here. People will fight and hiss over the least important of things, such as when the ball gag was removed, and the fire storm that ensued once round-start lizards were enabled. The "furry" thing is a red-hot topic in this community, I would say it is similar to the controversy surrounding LGBT. 

Most of the feedback will be useless to you, with the only counter argument I would see valid about this would be disguises.

Miauw62 commented:
*coughs*
if this is so shitty, than maybe we should consider that the same might be true for the ugly-ass lizard customization.

Kromgar commented:
>Comparing furries to LGBT

There's a difference one is a fetish and the other is your genetically predispositioned sexuality.

Steelpoint commented:
Lizard customization is shitty Miauw.

Kromgar commented:
>Comparing furries to LGBT

There's a difference one is a fetish and the other is your genetically predispositioned sexuality.

Dressing up in a fur suit and saying how you wish you were an Anthromorphic wolf is in no way fucking comparable to a person's sexuality thats pretty damn insulting Gunhog.

Ricotez commented:
@Kromgar your entire argument is invalid because your foundation is that this is some kind of creepy attempt to get my fetishes in-game.

<b>ERP is banned</b>. Why the fuck would I waste time on a pull request like this if it was about fetishes. What I want to add is more customization options, and Miauw made the actually valid counterargument that all the customization makes disguising very difficult, which upsets the balance of a game that's all about deceit and paranoia. That is the argument that actually makes me consider closing this PR for a second time.

duncathan commented:
Just leave it for the green buttons to decide.

IratePirate commented:
And I thought TG was the one with the supposedly rigorously enforced coder standards.

Steelpoint commented:
@IratePirate That's a funny joke, honestly we've never had any real enforced coder standards.

freerealestate commented:
is this attracting looky-loos from other codebases now?

killer653 commented:
:+1: 
what are you waiting for
do it
just
DO IT

NigglyWiggly commented:
@freerealestate 
>is this attracting looky-loos from other codebases now?

Y E S

Kromgar commented:
![image](https://cloud.githubusercontent.com/assets/6778786/8411983/df9d23e2-1e54-11e5-9814-bac2af9290c8.png)

palpatine213 commented:
Disguising as a lizard would be much harder than disguising as a cat person. If we're using that as reasoning to close the PR, then lizards should also be removed.
With cat people you can just steal a pair of kitty ears, with lizards there's no way to disguise as on. You can't even imitate one over the radio with a voice changer, as they have their snowflake speech patterns.

Ricotez commented:
@Kromgar 

<img src="http://38.media.tumblr.com/tumblr_m2rs3vze471rtcfaqo1_500.gif">

Rinoka commented:
If this is going to happen and we're all getting up in arms about the paranoia factor, let's figure out how to balance that shit.

1) Either remove saymods, or add a saymod determining menu to all voice changers. Otherwise, it is literally impossible to impersonate both lizards and catpeople based on what words are used to describe how they talk, which is basically literally what you would use a voice changer for.

2) Find a way to make tails availible to everyone. Surgically ripping them off their owners, adding a hook to them, and wearing them? Sure. Adding a secret hacked furry heaven in the clothesmate? Also sure. Also, let them be dyed whatever color; washing them with crayons or what have you.

Essentially, the only boons they give are fluff... but enough fluff to make anyone trust they're who they say they are. If you can make it so there's ways to disguise as them, I have no problem with this passing.

(Mainly because they will probably be beaten to death within a few minutes of roundstart.)

palpatine213 commented:
There are no saymods for cat people currently, only lizards.

duncathan commented:
@rinoka
Number two will probably be handled when Ricotez (coincidentally, the creator of this PR) finished dismemberment. Number one isn't applicable to cat persons, but should be dealt with whether this PR goes in or not.

Ricotez commented:
I have thought about adding a temporary surgery that lets you cut off human ears and tails, and simply merge it into the amputation surgery once the time for dismemberment comes.

Heck I could even add Incoming's catbeest on a stick recipe but with real ears, so you can feed someone their own ears on a stick.

Commit:
-Adds new proc-based job check for species. If enforce_human_authority is set, the various pieces of code that deal with roundstart job stuff will call this proc on your preferred species with 2 arguments: the name of the job and the list of mutantparts you have currently set.
-The proc is defined as proc/qualifies_for_rank(var/rank, var/list/features), and by default defined to always return 1. You must override it locally in your species definition to set special job properties.
-Lizards will fail the check for jobs in command_positions and pass it for any other job. Just like before.
-Humans will always pass the check if they don't have ears or tails. If they do have ears or tails, they will fail the check if the job is in command_positions, engineering_positions, science_positions, medical_positions, security_positions or equal to "Quartermaster", effectively limiting their job selection to civlian, lower supply and silicon jobs.

Xhuis commented:
\>incoming's
___I MADE THAT SHIT YOU SON OF A BITCH___

duncathan commented:
>if humans have ears or tails, they can't be command, engineering, science, medical, or sec
:+1: as long as they're objectively inferior, they're perfect. Otherwise people could have their snowflake and eat it too.

Ricotez commented:
sorry xhuis, so many people have been yelling at me that I kinda lost track

will you give me permission to include your catbeest recipe if I decide to go through with the surgery thing? I think that letting people feed catpeople their own ears will lead to a lot of shenanigans (griff)

Freshmaniscoolman commented:
@Ricotez I think you should kill yourself, you fucking gigantic niggerfaggot.

Ricotez commented:
@Freshmaniscoolman

<img src="https://www.101greatgoals.com/wp-content/uploads/2014/11/there-is-no-need-to-be-upset.png">

Freshmaniscoolman commented:
@Ricotez You think you're a cat, that's really all I have to say. You should seek mental help ASAP. I'm trying to help you buddy, don't think everybody's out there to stab you in the back. Ok? I got your back. It's ok sweetie.

duncathan commented:
@Freshmaniscoolman have you eve heard the term "ad hom"?

Freshmaniscoolman commented:
@duncathan Have you ever heard the term "Suck my dick"?

Ricotez commented:
@Freshmaniscoolman I'm sorry, but I think there's something wrong with your keyboard, all I can read in your posts is WAAH  WAAH IM A GIANT BABY WHO CAN ONLY BE COOL BY USING RACIST AND HOMOPHOBIC SLURS AND MAKING DEATH THREATS

duncathan commented:
@Freshmaniscoolman Yeah, it tends to be used when people don't have any legitimate arguments left, and they have to resort to attacking the character of their opponent to get a point accross. Which, by the way, falls under that "ad hom" thing I mentioned earlier.

Freshmaniscoolman commented:
@Ricotez Wow, I'm surprised you could read, that's exactly what I'm intending to send across. Faggot nigger I want to kill you. Now go masturbate to some cats, will you? fag.

Freshmaniscoolman commented:
@duncathan *tips fedora*

Kromgar commented:
The rage is real

duncathan commented:
This is absolutely hilarious

KorPhaeron commented:
>arguing with someone who is permabanned

Freshmaniscoolman commented:
https://cotocrew.files.wordpress.com/2015/01/10c.jpg

phil235 commented:
Stop shitposting here. I'm deleting every post not relevant and useful.

Ricotez commented:
image he just posted is extremely graphic. all you can expect from someone who is mad enough to shitpost in a PR despite having a permaban.

tip of the round: try it with a cool dog.

<img src="http://www.iwantcovers.com/wp-content/uploads/2012/03/Cool-Dog.jpg">

Cheridan commented:
There's a point where this needs to stop and we've clearly passed it. Safe to say that cat ear and tail options are, erm, contentious. The code you've got here looks like a good framework for cosmetics and species-based limitations and can get merged but for use down the line. However the particular cosmetics you have chosen aren't suitable for the purposes of this codebase at the time.

Commit:
-Removed roundstart access to ears and human tails. They are an admin only thing now.
-You can now set which colour an accessory should use: HAIR, FACEHAIR, EYECOLOR, MUTCOLORS or its own (0). Right now only mutantparts use this.
-Consequently, humans lost their ability to change mutant colours, although the special flag MUTCOLORS_PARTSONLY still exists.

Cheridan commented:
haha funny joke @incoming

Cheridan commented on code/modules/mob/living/carbon/human/species.dm:
```diff
@@ -294,12 +316,33 @@
+			if(bodypart == "tail_lizard" || bodypart == "tail_human")
+				bodypart = "tail"
+			else if(bodypart == "waggingtail_lizard" || bodypart == "waggingtail_human")
+				bodypart = "waggingtail"
```
This could probably be done better. *bodypart = tail_[species]* instead of || checks.

Ricotez replied:
The idea of tail_lizard and tail_human isn't to distinguish between two species, but between two different tail lists. If another species wants to use lizard or human tails, they wouldn't be able to if I did this.

It's not super elegant but the alternative was defining exactly which individual accessories every species can pick and I figured that there are enough differences between lizard and mammal tails to warrant two lists.

Cheridan replied:
Ah, ok.

freerealestate replied:
Making content admin-only is terrible imo

Just leads to favouritism and self-serving.

On 1 July 2015 at 04:04, Cheridan <notifications@github.com> wrote:

> In code/modules/mob/living/carbon/human/species.dm
> <https://github.com/tgstation/-tg-station/pull/10209#discussion_r33590112>
> :
>
> >  				if("body_markings")
> >  					S = body_markings_list[H.dna.features["body_markings"]]
> >
> >  			if(!S || S.icon_state == "none")
> >  				continue
> > +
> > +			//A little rename so we don't have to use tail_lizard or tail_human when naming the sprites.
> > +			if(bodypart == "tail_lizard" || bodypart == "tail_human")
> > +				bodypart = "tail"
> > +			else if(bodypart == "waggingtail_lizard" || bodypart == "waggingtail_human")
> > +				bodypart = "waggingtail"
>
> Ah, ok.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tgstation/-tg-station/pull/10209/files#r33590112>.
>

RemieRichards commented:
Well look at the latest commit, it's admin only, not roundstart now, SURELY that's enough of a change.

On the topic of the deleted comments on "muh code standards"
Guess what? <i>the code is good</i> get your heads out of your asses.

RemieRichards commented on code/modules/mob/living/carbon/human/species.dm:
```diff
@@ -294,12 +316,33 @@
 				continue
+
+			//A little rename so we don't have to use tail_lizard or tail_human when naming the sprites.
+			if(bodypart == "tail_lizard" || bodypart == "tail_human")
```
same as cheri's comment below

RemieRichards commented:
Unlocked by request of author.

PKPenguin321 commented:
if this stuff stays admin only then i guess i'm okay with it now

GunHog commented:
Why admin only? :(  Will it be changed in the future? :/

PKPenguin321 commented:
GunHog please go

killer653 commented:
If it's admin only, there's no reason not to have it in the game. Admins could punish/reward someone with them. There's really no problem in that.

Rinoka commented:
Basically. Roundstart meant a lot of balance stuff would be needed. Admingifted is much better until there's a bit more system balance. :+1:

Commit:
-Added new option to the VV menu drop-down list: Toggle Purrbation.
-Putting a human on purrbation gives them cat ears and a cat tail, and sends them the message 'You suddenly feel valid.'
-Removing a human from purrbation reverses the effects, and sends them the message 'You suddenly don't feel valid anymore.'
-Renamed "Long and Smooth" tail to "Cat" tail. I wasn't kitten anyone with the old name.
-Updated changelog.
Forgot to include these two in the last commit:
-Cat ears and tail now use your hair colour instead of your mutant colour.
-Added changelog.

Ricotez commented:
This last commit will probably be the final one for this PR. By the way, I forgot to include these two parts with it:
-Commented out the saving of your ear and tail_human preferences. It's unneeded bloat of save files right now.
-Re-added the setup window stuff for choosing ears and human tails, but commented them out. This makes it easier for other servers to include them.

optimumtact commented:
Don't comment them out

Wrap them in an define tag and leave commented out compile options in /code/_compile_options.dm

Example here
https://github.com/tgstation/-tg-station/blob/master/code/__HELPERS/bygex/bygex.dm#L26

Ricotez commented:
Don't we have the config file for that stuff? I could add a config option

Commit:
- Added new confic option: JOIN_WITH_MUTANT_HUMANS
- Uncommented all stuff that was commented in the last commit.
- Added a check in front of it that blocks them from showing up in the setup window if JOIN_WITH_MUTANT_HUMANS is not set.
- Included new game_options.txt as an example.

The tgstation server has never used the example game_options.txt so the new config option will not be enabled on it. Even if it did, the option starts commented out in the file.

Ricotez commented:
I should have just gone with a config option right away, it's much more elegant than commenting out parts of the code.

For the record: the config option is not set by default, especially not on the server because it uses its own config files. It's a matter of policy now, and the policy currently appears to be "cats are valid" and "round start? hell no".

optimumtact commented:
Yeah that works :+1:

Stickymayhem commented:
If this became admin only I'd deadmin anyone who used it faster than if they had possessed a level 6 singularity

RemieRichards commented:
@Stickymayhem abuse of power.

Kelenius commented:
Why so much hate?

KorPhaeron commented:
Editing cat ears onto Jack Steiner every round

Ricotez commented:
I have no idea why this receives so much hate, but when half the community speaks out against something, I'm not going to ignore it. A lot more people agree to it if it's admin only, but it's a shame to just throw away code that may be of use to other servers. Therefore it's now behind a config lock.

@Stickymayhem 

<img src="http://33.media.tumblr.com/766e88e17de73f14eeb7c99b45cc1e7c/tumblr_nagsbhPU3x1txrzruo1_500.gif" />

Iamgoofball commented:
@Stickymayhem can I take this as permission to hellfoam people with genetic cattebeest ears and tails without consequence

Nienhaus commented:
:+1:

Razharas commented:
Is the shit done?

Ricotez commented:
If you guys accept it, I have nothing left to add.

Chocobro commented:
total ideas for the framework include: scars, hair tints, wounds, freckles, beauty spots, makeup, writing, tattoos, crumbs, missing minor limbs (ie. a metal ass), skin discolourations, crow's feet.
these almost definitely won't be in this PR, but it's good to list them in case anyone plans on adding stuff.

palpatine213 commented:
Merged when?

Ricotez commented:
as soon as a maintainer sees this

Razharas commented:
Its ok

Steelpoint commented:
Where were you when the world ended.

duncathan commented:
It begins.

palpatine213 commented:
>Cat parts default to ON! Hahahahahaha

Incoming5643 commented:
Actually they don't default to on, they don't default to anything at all when they've never been enabled, and the check only looks for "none" and not null.

Fixing with rapid speed

RogerWilcoBeep commented:
NOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO

Miauw62 commented:
l o l

SkowronX commented:
dohoho

d3athrow commented:
:^3 what kind of fursona do you guys have, i identify as a chicken! Wattles and Combs when?????

Iamgoofball commented:
#AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA

duncathan commented:
is that entirely necessary

Iamgoofball commented:
# yes

Incoming5643 commented:
<img src="https://33.media.tumblr.com/tumblr_m3qfx05FYt1qkbonq.gif"></src>

Ricotez commented:
stop shitposting in this PR even after it got merged, I have a fix up at #10353

it was a simple problem that was easy to overlook while testing and it had a simple solution

PKPenguin321 commented:
you can tell by the bugs that were introduced with this that it was tested thoroughly and is a quality and extremely necessary addition to our codebase

palpatine213 commented:
Said bugs likely occurred due to the drama surrounding the PR distracting people.

IratePirate commented:
Incredible. Can we port this?

francinum commented:
@IratePirate AGPL Licence.

RemieRichards commented:
@francinum @IratePirate is probably asking for a re-licence, don't have to worry about AGPL if the creator of the code says "for vg it's GPL" and lets them port it.

I've done it on request before.

Ricotez commented:
@IratePirate SInce this code got merged it's not like I can forbid it anyway, but you have my personal approval too. However you should realize that right now this code is bugged in 2 ways:
- Gives pre-existing character save files a 75% chance to have at least one of the two new features.
- Does not save tail_human or ears preference, so the first bug happens every time.
I've fixed this in #10353 (also adds adminlogging for the "Toggle Purrbation" feature) but that one is not yet merged so you'll have to port it separately.

RemieRichards commented:
@Ricotez It's totally within your rights to deny them a re-licence.
Just because it was merged here doesn't mean it can or should be elsewhere, it's 100% up to you to hand out that right as the author. handing out a re-licence is akin to their being 2 versions, one AGPL for ourselves which we have merged, and an "extra" one that's GPL for vg.

I'm aware you've said yes as-well but I thought I'd clear it up.

Ricotez commented:
@RemieRichards I'm fine with people porting my stuff, but I extended Incoming's system and split up the existing cat ears sprite so the amount of original work in this is actually quite small

RemieRichards commented:
@Ricotez again, I'm just clarifying your rights as an author.

Ricotez commented:
#10353

<img src="http://38.media.tumblr.com/0b45ade95a61fba4b5533ec8d8107612/tumblr_nql0d7VYVF1s8sotmo1_540.gif" />

fix got merged

Incoming5643 commented:
For the record I'm fine with ports to other codebases too when it comes to things I made. It's all part of the community.

RemieRichards commented:
So am I, I was just clarifying as Rico's response felt like he was saying it because he has too.

GunHog commented:
@Ricotez from what anime is that gif above? Oh, and since there is a permission train going here, I am perfectly fine with people using any code I commit to /tg/ as well.

Ricotez commented:
@GunHog it's from Yozakura Quartet

francinum commented:
Round it out to an even 200.

francinum commented:
Once more.

optimumtact commented:
I can't believe you've done this.

SconesC commented:
beginning of the end

PKPenguin321 commented:
:ok_hand::ok_hand::ok_hand::ok_hand::ok_hand::ok_hand::ok_hand::ok_hand::ok_hand::ok_hand::ok_hand::ok_hand::ok_hand::ok_hand::ok_hand::ok_hand::ok_hand::ok_hand::ok_hand::ok_hand:
